src/supremm/datasource/pcp/pcpsummarize.py

```diff
@@ -0,0 +1,365 @@
+        # see the original data
+        self.rangechange.set_fetched_metrics([])
+
+        if len(metric_id_array) == 0:
+            logging.debug("Skipping %s (%s)" % (type(preproc).__name__, preproc.name))
+            preproc.hostend()
+            return
+
+        mtypes = pcpcinterface.getmetrictypes(ctx, metric_id_array)
+
+        done = False
+
+        while not done:
+            try:
+                result = ctx.pmFetch(metric_id_array)
+
+                if False == self.runpreproccall(preproc, result, mtypes, ctx, mdata, metric_id_array):
+                    # A return value of false from process indicates the computation
+                    # failed and no more data should be sent.
+                    done = True
+
+                ctx.pmFreeResult(result)
+
+            except pmapi.pmErr as exp:
+                if exp.args[0] == c_pmapi.PM_ERR_EOL:
+                    done = True
+                else:
+                    preproc.status = "failure"
+                    preproc.hostend()
+                    raise exp
+            except Exception as exp:
+                preproc.status = "failure"
+                preproc.hostend()
+                raise exp
+
+
+        preproc.status = "complete"
+        preproc.hostend()
+
+    def processforanalytic(self, ctx, mdata, analytic):
+        """ fetch the data from the archive, reformat as a python data structure
+        and call the analytic process function """
+
+        metric_id_array, metricnames = pcpcinterface.getmetricstofetch(ctx, analytic)
+
+        if len(metric_id_array) == 0:
+            logging.debug("Skipping %s (%s)" % (type(analytic).__name__, analytic.name))
+            return
+
+        self.rangechange.set_fetched_metrics(metricnames)
+
+        mtypes = pcpcinterface.getmetrictypes(ctx, metric_id_array)
+
+        done = False
+
+        while not done:
+            result = None
+            try:
+                result = ctx.pmFetch(metric_id_array)
+
+                if False == self.runcallback(analytic, result, mtypes, ctx, mdata, metric_id_array):
+                    # A return value of false from process indicates the computation
+                    # failed and no more data should be sent.
+                    done = True
+
+            except pmapi.pmErr as exp:
+                if exp.args[0] == c_pmapi.PM_ERR_EOL:
+                    done = True
+                else:
+                    logging.warning("%s (%s) raised exception %s", type(analytic).__name__, analytic.name, str(exp))
+                    analytic.status = "failure"
+                    raise exp
+            finally:
+                if result != None:
+                    ctx.pmFreeResult(result)
+
+        analytic.status = "complete"
+
+    def logerror(self, archive, analyticname, pmerrorcode):
+        """
+        Store the detail of archive processing errors
+        """
+        #if analyticname != "gpu":
```

Comment:
remove commented out code.

Reply:
Resolved in https://github.com/ubccr/supremm/pull/259/commits/a50f9f32a7e63d5804f3f01844149b6cc2936be9